packages/table/src/quadrants/tableQuadrantStack.tsx

```diff
@@ -0,0 +1,438 @@
+     */
+    columnHeaderRef?: (ref: HTMLElement) => void;
+
+    /**
+     * The grid computes sizes of cells, rows, or columns from the
+     * configurable `columnWidths` and `rowHeights`.
+     */
+    grid: Grid;
+
+    /**
+     * Whether horizontal scrolling is currently disabled.
+     * @default false
+     */
+    isHorizontalScrollDisabled?: boolean;
+
+    /**
+     * If `false`, hides the row headers and settings menu.
+     * @default true
+     */
+    isRowHeaderShown?: boolean;
+
+    /**
+     * Whether vertical scrolling is currently disabled.
+     * @default false
+     */
+    isVerticalScrollDisabled?: boolean;
+
+    /**
+     * The number of frozen columns.
+     */
+    numFrozenColumns?: number;
+
+    /**
+     * The number of frozen rows.
+     */
+    numFrozenRows?: number;
+
+    /**
+     * An optional callback invoked the quadrants are scrolled.
+     */
+    onScroll?: React.EventHandler<React.SyntheticEvent<HTMLElement>>;
+
+    /**
+     * A callback that receives a `ref` to the main-quadrant element.
+     */
+    quadrantRef?: (ref: HTMLElement) => void;
+
+    /**
+     * A callback that renders either all of or just frozen sections of the table body.
+     */
+    renderBody: (
+        quadrantType: QuadrantType,
+        showFrozenRowsOnly?: boolean,
+        showFrozenColumnsOnly?: boolean,
+    ) => JSX.Element;
+
+    /**
+     * A callback that renders either all of or just the frozen section of the column header.
+     */
+    renderColumnHeader?: (showFrozenColumnsOnly?: boolean) => JSX.Element;
+
+    /**
+     * A callback that renders the table menu (the rectangle in the top-left corner).
+     */
+    renderMenu?: (refHandler: (ref: HTMLElement) => void) => JSX.Element;
+
+    /**
+     * A callback that renders either all of or just the frozen section of the row header.
+     */
+    renderRowHeader?: (refHandler: (ref: HTMLElement) => void, showFrozenRowsOnly?: boolean) => JSX.Element;
+
+    /**
+     * A callback that receives a `ref` to the main quadrant's row-header container.
+     */
+    rowHeaderRef?: (ref: HTMLElement) => void;
+
+    /**
+     * A callback that receives a `ref` to the main quadrant's scroll-container element.
+     */
+    scrollContainerRef?: (ref: HTMLElement) => void;
+}
+
+export class TableQuadrantStack extends AbstractComponent<ITableQuadrantProps, {}> {
+    // we want the user to explicitly pass a quadrantType. define defaultProps as a Partial to avoid
+    // declaring that and other required props here.
+    public static defaultProps: Partial<ITableQuadrantProps> & object = {
+        isHorizontalScrollDisabled: false,
+        isRowHeaderShown: true,
+        isVerticalScrollDisabled: false,
+    };
+
+    private quadrantRefs = {
+        [QuadrantType.MAIN]: {} as IQuadrantRefs,
+        [QuadrantType.TOP]: {} as IQuadrantRefs,
+        [QuadrantType.LEFT]: {} as IQuadrantRefs,
+        [QuadrantType.TOP_LEFT]: {} as IQuadrantRefs,
```

Comment:
`QuadrantType` needs to be a string: 
![image](https://user-images.githubusercontent.com/443450/28682848-4d54f2a2-72b3-11e7-98c2-63649cdc5c8e.png)

And can't do an interface, because having `QuadrantType` keys yields an error: `A computed property name in an interface must directly refer to a built-in symbol`.